recipes/usrsctp/all/conanfile.py

```diff
@@ -1,74 +1,85 @@
-    homepage = "https://github.com/sctplab/usrsctp"
     url = "https://github.com/conan-io/conan-center-index"
-    topics = ("conan", "network", "sctp")
-    description = " A portable SCTP userland stack"
-    settings = "os", "compiler", "arch", "build_type"
-    options = {"shared": [True, False],
-               "fPIC": [True, False]}
-    default_options = {"shared": False,
-                       "fPIC": True}
-    exports_sources = ["CMakeLists.txt", "patches/*"]
-    generators = "cmake"
+    homepage = "https://github.com/sctplab/usrsctp"
+    topics = ("network", "sctp")
 
-    _cmake = None
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "usrsctp-{}".format(self.version)
-        os.rename(extracted_dir, self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["sctp_debug"] = False
+        tc.variables["sctp_werror"] = False
+        tc.variables["sctp_build_shared_lib"] = self.options.shared
+        tc.variables["sctp_build_programs"] = False
+        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = self.options.shared
+        tc.generate()
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["sctp_debug"] = False
-        self._cmake.definitions["sctp_werror"] = False
-        self._cmake.definitions["sctp_build_shared_lib"] = self.options.shared
-        self._cmake.definitions["sctp_build_programs"] = False
-        self._cmake.configure()
-        return self._cmake
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        # Fix "The CMake policy CMP0091 must be NEW, but is ''"
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
```

Comment:
We might want to report this upstream so that the version check comes before `project`. Are you up for that? :)